tests/unit/ml_handlers/test_spacy.py

```diff
@@ -0,0 +1,170 @@
+
+        self.run_sql(
+            """
+            CREATE MODEL proj.spacy__dependency_parsing__model
+            PREDICT recognition
+            USING
+              engine = 'spacy'
+              linguistic_feature = 'dependency-parsing',
+              target_column = 'text';
+            """
+        )
+
+        result_df = self.run_sql(
+            """
+            SELECT recognition
+            FROM proj.spacy__dependency_parsing__model
+            WHERE
+            text='Apple is looking at buying U.K. startup for $1 billion';
+            """
+        )
+
+        assert '{('"', \'punct', 'looking', 'VERB', '[]'), ('for', 'prep', 'startup', 'NOUN', '[billion]'), ('1', 'compound', 'billion', 'NUM', '[]'), ('Apple', 'nsubj', 'looking', 'VERB', '[]'), ('buying', 'pcomp', 'at', 'ADP', '[U.K.]'), ('looking', 'ROOT', 'looking', 'VERB', '[\", Apple, is, at, startup, \"]'), ('$', 'quantmod', 'billion', 'NUM', '[]'), ('is', 'aux', 'looking', 'VERB', '[]'), ('billion', 'pobj', 'for', 'ADP', '[$, 1]'), ('startup', 'dep', 'looking', 'VERB', '[for]'), ('at', 'prep', 'looking', 'VERB', '[buying]'), ('U.K.', 'dobj', 'buying', 'VERB', '[]')}" in result_df["recognition"].iloc[0]
+
+    @patch("mindsdb.integrations.handlers.postgres_handler.Handler")
+    def test_spacy_pos_tagging(self, mock_handler):
+        self.run_sql("CREATE DATABASE proj")
+
+        text = ["Apple is looking at buying U.K. startup for $1 billion"]
+        df = pd.DataFrame(text, columns=['text'])
+
+        self.set_handler(mock_handler, name="pg", tables={"df": df})
+
+        self.run_sql(
+            """
+            CREATE MODEL proj.spacy__pos_tag__model
+            PREDICT recognition
+            USING
+              engine = 'spacy'
+              linguistic_feature = 'pos-tag',
+              target_column = 'text';
+            """
+        )
+
+        result_df = self.run_sql(
+            """
+            SELECT recognition
+            FROM proj.spacy__pos_tag__model
+            WHERE
+            text='Apple is looking at buying U.K. startup for $1 billion';
+            """
+        )
+
+        assert "{('startup', 'startup', 'NOUN', 'NN', 'dep', 'xxxx', True, False), ('buying', 'buy', 'VERB', 'VBG', 'pcomp', 'xxxx', True, False), ('U.K.', 'U.K.', 'PROPN', 'NNP', 'dobj', 'X.X.', False, False), ('1', '1', 'NUM', 'CD', 'compound', 'd', False, False), ('for', 'for', 'ADP', 'IN', 'prep', 'xxx', True, True), ('$', '$', 'SYM', '$', 'quantmod', '$', False, False), ('Apple', 'Apple', 'PROPN', 'NNP', 'nsubj', 'Xxxxx', True, False), ('billion', 'billion', 'NUM', 'CD', 'pobj', 'xxxx', True, False), ('"', '"', 'PUNCT', "''", 'punct', '"', False, False), (\'looking\', \'look\', \'VERB\', \'VBG\', \'ROOT\', \'xxxx\', True, False), ('"', '"', \'PUNCT\', \'``\', \'punct\', '"', False, False), ('at', 'at', 'ADP', 'IN', 'prep', 'xx', True, True), ('is', 'be', 'AUX', 'VBZ', 'aux', 'xx', True, True)}" in result_df["recognition"].iloc[0]
```

Comment:
Perhaps there could be an "output parser" option where the resulting DF is easier to grasp than these nested results.

I get how the current behavior is what we want for batched predictions, but it feels like we could improve the structure for one-off cases. Thoughts?

Reply:
Right so for one-off cases, you would like to spread the attributes of pos-tag to DF columns. Is this what you mean? We could do that for one-off cases.

Reply:
I was thinking about this and it would provide users with more clarity and faster examination of words-to-[pos-tag]. I would also spread the sentences to words as a row in DF, this way we know which word relates to the pos-tag. Maybe create a boolean if the users want to spread sentences to words. I will do that for the next PR.

Reply:
Both sound good! Can we include it in this PR though?

Reply:
Working on it!